src/sidebar/components/test/group-list-test.js

```diff
@@ -363,6 +385,89 @@ describe('groupList', function() {
       '_blank'
     );
   });
+  describe('group section visibility', () => {
+    [
+      {
+        description:
+          'shows Currently Viewing section when there are currently viewing groups',
+        currentlyViewingGroups: [publicGroup],
+        featuredGroups: [restrictedGroup],
+        myGroups: [],
+        expectedSections: ['Currently Viewing', 'Featured Groups'],
+      },
+      {
+        description:
+          'shows Featured Groups section when there are featured groups',
+        currentlyViewingGroups: [],
+        featuredGroups: [restrictedGroup],
+        myGroups: [publicGroup],
+        expectedSections: ['Featured Groups', 'My Groups'],
+      },
+      {
+        description: 'shows My Groups section when there are my groups',
+        currentlyViewingGroups: [],
+        featuredGroups: [],
+        myGroups: [publicGroup, privateGroup],
+        expectedSections: ['My Groups'],
+      },
+    ].forEach(
+      ({
+        description,
+        currentlyViewingGroups,
+        featuredGroups,
+        myGroups,
+        expectedSections,
+      }) => {
+        it(description, () => {
+          fakeFeatures.flagEnabled.withArgs('community_groups').returns(true);
+          // In order to show the group drop down there must be at least two groups.
+          groups = currentlyViewingGroups
+            .concat(featuredGroups)
+            .concat(myGroups);
+          fakeStore.getCurrentlyViewingGroups = sinon
+            .stub()
+            .returns(currentlyViewingGroups);
+          fakeStore.getFeaturedGroups = sinon.stub().returns(featuredGroups);
+          fakeStore.getMyGroups = sinon.stub().returns(myGroups);
```

Comment:
This is fine, but if something is already a stub you can just call `returns(...)` on the existing instance to change the result.